src/parser/mod.rs

```diff
@@ -15064,6 +15117,38 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    /// Parse [Statement::Go]
+    fn parse_go(&mut self) -> Result<Statement, ParserError> {
+        self.expect_previously_only_whitespace_until_newline()?;
+
+        let count = loop {
+            // using this peek function because we want to halt this statement parsing upon newline
```

Comment:
Can we include the example you had in the comment earlier, explicitly highlighting why this statement is special? I think otherwise it would not be obvious to folks that come across the code later on why the current code is a special case

Reply:
Yes, I can add that comment. However there is also test coverage for this behavior, so it should be protected from future refactoring

Reply:
Done 👍